Added initial CapEx breakdown to ProjectManager.

JakeNunemaker opened the pull request:
@gbarter Is this what you had in mind? This level of breakdown (by module) is the easiest to implement. See below for the API and how the results are structured.

![Screen Shot 2021-03-16 at 10 10 24 AM](https://user-images.githubusercontent.com/42254629/111344327-e6510e00-8641-11eb-8e67-0deb6fbd0db4.png)

JakeNunemaker commented:
Closes #91

Matt-Shields commented on ORBIT/manager.py:
```diff
@@ -1114,6 +1126,61 @@ def installation_capex_per_kw(self):
+        unique = np.unique(
+            [*self.system_costs.keys(), *self.installation_costs.keys()]
+        )
+        categories = {}
```
Will this pull from custom design/install modules?

JakeNunemaker replied:
This will pull costs from custom modules but with the way it is coded up right now they would be classified as 'Misc' without some hacking. I had an idea at lunch that may resolve this though.

JakeNunemaker replied:
I revised this a bit so that `ProjectManager` pulls the capex category from individual modules so it'll work with custom modules now. 

In custom modules, you can now define the category like this:

```
class CustomModule(InstallPhase):
    
    capex_category = "Substructure"
    expected_config = {...}
    
    def __init__(self, ):
        ...
```

If it doesn't find capex_category it'll print a warning and put the costs in 'Misc.'

Matt-Shields reviewed (approved):
Love it!

gbarter commented:
This is definitely the info I was looking for the other day.  Thank you!

@Matt-Shields : Is it worth pulling out the ports and planning costs separately as the JEDI folks had requested?

@JakeNunemaker : My other difficulty is that the different summary outputs are accessed via `project.magic_keyword` and I have to go hunting for the exact magic keyword.  On the inputs, ORBIT uses a global config dictionary.  Maybe there could be a global output dictionary too?  Does that make sense?

JakeNunemaker commented:
@gbarter Port costs are rolled into the "Component Installation" subcategories above. That is a good point about the soft/planning costs though, I can add those in there as well.

For the output dict, do you want all of these costs rolled into one output keyword. For example,

```
project.outputs = {
    'total_capex': 1e9,
    'total_capex_per_kw': 1500,
    'capex_breakdown': dict,
    'capex_breakdown_per_kw': dict
    'soft_capex': num,
    'soft_capex_breakdown': dict
}
```

This would contain all of the main outputs at varying levels of aggregation per MW equivalents.

gbarter commented:
Yes, `project.output={...}` would be very helpful!  I am okay with some entries appearing in multiple categories, but I understand there could be room for misinterpretation there.  

If you are toying with the outputs, another useful data dump, perhaps only activated with a user flag, would be to write the detailed event log to an Excel file.  I have seen you do really impressive and detailed analyses with that log, but it is really only accessible if you know where to get it and are a Pandas black belt.  Some of the industry users might enjoy bringing it into excel to work with.

JakeNunemaker commented:
@gbarter @Matt-Shields  I added a `project.outputs()` method. I didn't make it a property so that I could add a couple options as well. By default it doesn't output the project logs or detailed npv outputs as they are long lists that clog up the stdout and make the method not very useful for seeing what is available. 

![Screen Shot 2021-03-18 at 9 45 26 AM](https://user-images.githubusercontent.com/42254629/111654895-c864e400-87ce-11eb-892c-5f113602f07f.png)

I also added a `export_project_logs` method that saves a .csv of the project logs.

gbarter commented:
This is perfect!
